pkg/service/elbv2.go

```diff
@@ -1,40 +1,57 @@
 package service
 
 import (
-	"context"
+	"errors"
+	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/service/elbv2"
 	"github.com/aws/aws-sdk-go/service/elbv2/elbv2iface"
 
 	"github.com/keikoproj/lifecycle-manager/pkg/log"
 )
 
-func waitForDeregisterTarget(elbClient elbv2iface.ELBV2API, arn, instanceID string, port int64) error {
+func waitForDeregisterTarget(event *LifecycleEvent, elbClient elbv2iface.ELBV2API, arn, instanceID string, port int64) error {
 	var (
-		MaxAttempts = 500
+		DelayIntervalSeconds int64 = 30
+		MaxAttempts                = 500
+		found                bool
 	)
 
-	waiterOpts := []request.WaiterOption{
-		request.WithWaiterMaxAttempts(MaxAttempts),
-	}
-
 	input := &elbv2.DescribeTargetHealthInput{
 		TargetGroupArn: aws.String(arn),
-		Targets: []*elbv2.TargetDescription{
-			{
-				Id:   aws.String(instanceID),
-				Port: aws.Int64(port),
-			},
-		},
 	}
 
-	err := elbClient.WaitUntilTargetDeregisteredWithContext(context.Background(), input, waiterOpts...)
-	if err != nil {
-		return err
+	for i := 0; i < MaxAttempts; i++ {
+
+		if event.eventCompleted {
+			return errors.New("event finished execution during deregistration wait")
+		}
+
+		found = false
+		targets, err := elbClient.DescribeTargetHealth(input)
+		if err != nil {
+			return err
+		}
+		for _, targetDescription := range targets.TargetHealthDescriptions {
+			if aws.StringValue(targetDescription.Target.Id) == instanceID && aws.Int64Value(targetDescription.Target.Port) == port {
+				found = true
+				if aws.StringValue(targetDescription.TargetHealth.State) == elbv2.TargetHealthStateEnumUnused {
+					return nil
+				}
+				break
+			}
+		}
+		if !found {
+			log.Debugf("target %v not found in target group %v", instanceID, arn)
+			return nil
+		}
+		log.Debugf("target %v is still deregistering from %v", instanceID, arn)
```

Comment:
same comment as in `elb.go` file
```suggestion
		log.Debugf("target %v is still deregistering from %v (%v/%v)", instanceID, arn, i, MaxAttempts)
```

Reply:
same